### PR TITLE
Support log for ansible-disaster-recovery

### PR DESCRIPTION
**Pull request by maorlipchuk**
1. Changes in vault and Configparse usages to avoid any missing dependecies in RHEL installation.
2. dr-play.yml will be copied to "files" folder to be used for the ovirt-dr script.
3. Introduce the use of log file in ovirt-dr script.
The script default location and level is configured in the dr.conf
Default location is /var/log/ovirt-dr/ovirt-dr.log and default level is "DEBUG"
The user can override it with --log-file and --log-level
If no log file will be specified stdout will be used to print the output.

### CODE REVIEW

**Comment by machacekondra** [comment]
Why not do it inside `build.sh` script? I've created it so it's simpler then using `Makefile`.

File: build.sh [hunk 1]
```diff
@@ -6,12 +6,14 @@ RPM_RELEASE="0.1.$MILESTONE.$(date -u +%Y%m%d%H%M%S)"
 
 ROLE_NAME="oVirt.disaster-recovery"
 PACKAGE_NAME="ovirt-ansible-disaster-recovery"
-PREFIX=/usr/local
+LOG_NAME="ovirt-dr"
+PPREFIX=/usr/local
```
Comment (machacekondra) [inline build.sh]: Why it's renamed to PPREFIX? I think it's typo, no?

File: build.sh [hunk 2]
```diff
@@ -38,11 +40,15 @@ install() {
...
   mkdir -p $PKG_DATA_DIR
   mkdir -p $PKG_DOC_DIR
+  mkdir -p $PKG_LOG_DIR/$LOG_NAME
+  touch $PKG_LOG_DIR/$LOG_NAME/$LOG_NAME.log
 
   cp -pR defaults/ $PKG_DATA_DIR
   cp -pR library/ $PKG_DATA_DIR
   cp -pR meta/ $PKG_DATA_DIR
   cp -pR tasks/ $PKG_DATA_DIR
+  cp -pR examples/ $PKG_DATA_DIR
```
Comment (machacekondra) [inline build.sh]: We store examples in /usr/share/doc for other roles, jut FYI, if you want to have it also in /usr/share/ you can, up to you.

Comment (maorlipchuk) [reply build.sh]: I would want to use the ansible play there in the script, that is why I'm copying it.

File: build.sh (hunk 1 above)
Comment (maorlipchuk) [reply build.sh]: ohh damn, you are right, will fix

**Comment by machacekondra** [comment]
@mwperina

File: build.sh [hunk 3]
```diff
@@ -6,12 +6,14 @@ RPM_RELEASE="0.1.$MILESTONE.$(date -u +%Y%m%d%H%M%S)"
 
 ROLE_NAME="oVirt.disaster-recovery"
 PACKAGE_NAME="ovirt-ansible-disaster-recovery"
+LOG_NAME="ovirt-dr"
```
Comment (mwperina) [inline build.sh]: If you are going to put logs into /var/log/ovirt-dr then please provide logrotate setup similarly as we do in engine:

https://github.com/oVirt/ovirt-engine/blob/master/packaging/sys-etc/logrotate.d/ovirt-engine.in

Comment (mwperina) [reply build.sh]: You also need to add dependency to logrotate, replace relative paths inside it (please take a look at GENERATED list and its usage in engine Makefile [1]) and install it into /etc/logrotate.d [2].

[1] https://github.com/oVirt/ovirt-engine/blob/master/Makefile#L211
[2] https://github.com/oVirt/ovirt-engine/blob/master/ovirt-engine.spec.in#L990

File: logrotate.d/ovirt-dr.in [hunk 4]
```diff
@@ -0,0 +1,8 @@
+"@PKG_LOG_DIR/LOG_NAME@/ovirt-dr.log" {
```
Comment (mwperina) [inline logrotate.d/ovirt-dr.in]: Where are you replacing `@PKG_LOG_DIR/LOG_NAME@` with `/var/log/ovirt-dr`? Also why is there only single log file? Where are you going to log multiple concurrent invocations of a role from single machine?

File: ovirt-ansible-disaster-recovery.spec.in [hunk 5]
```diff
@@ -26,10 +26,12 @@ This Ansible role provide funtionality to perform disaster recovery for oVirt en
 export PKG_DATA_DIR_ORIG=%{_datadir}/%{ansible_roles_dir}/%{roleprefix}%{rolename}
 export PKG_DATA_DIR=%{buildroot}$PKG_DATA_DIR_ORIG
 export PKG_DOC_DIR=%{buildroot}%{_pkgdocdir}
+install -dm 755 "$PKG_LOG_DIR/$LOG_NAME/$LOG_NAME.log"
```
Comment (mwperina) [inline ovirt-ansible-disaster-recovery.spec.in]: Please add `Requires: logrotate` spec file

Comment (maorlipchuk) [reply ovirt-ansible-disaster-recovery.spec.in]: done

File: logrotate.d/ovirt-dr.in (hunk 4 above)
Comment (maorlipchuk) [reply logrotate.d/ovirt-dr.in]: That is what I've defined in the spec, I will try to do that with '/var/log/ovirt-dr'
I've tried to follow how ovirt-engine works, should I do it like this:
  "/var/log/ovirt-dr"/*.log

Comment (maorlipchuk) [reply logrotate.d/ovirt-dr.in]: Is it ok if the file is located here:  logrotate.d/ovirt-dr.in

File: ovirt-ansible-disaster-recovery.spec.in (hunk 5 above)
Comment (mwperina) [reply ovirt-ansible-disaster-recovery.spec.in]: You need to use `Requires: logrotate`, logrotate is a package name

**Comment by mwperina** [comment]
> Thinking of it, I think the best approach for DR would be to have --log-file and --log-level command line options, instead of persistent log directory maybe it will be even more simple. But I will leave this decision on you.

+1
Unless role is running from a daemon or some other periodical stuff, then above is much better solution

**Comment by maorlipchuk** [comment]
I see, so I get a few questions to clarify:
1) If we will not have a persistent log directory, where should be the default log target file/dir?
2) Will every script runing should override the existing log which was before? Should we add somekind of validation whether the file already exists or not?
3) Is there any reflection on the build.sh or spec that I need to take care of?

Thanks

**Comment by machacekondra** [comment]
> Will every script runing should override the existing log which was before? Should we add somekind of validation whether the file already exists or not?

You need to create it if it don't exist. If should append to it if it exist.

> Is there any reflection on the build.sh or spec that I need to take care of?

Nope, no changes in `build.sh` script.

**Comment by maorlipchuk** [comment]
I've been tried to use the stdout as default if no log-file is specified but that really makes the output unreadable.
I've used the dr.conf to make a default location for a log file instead, if the user will desire to use the stdout he can always remove the default value from the dr.conf
The default location is /var/log/ovirt-dr/ovirt-dr.log, feel free to suggest any other place if you think it will be more appropriate.
Let me know what you think, whether you prefer to make the default as stdout or leave it as it is with a log file

**Comment by machacekondra** [comment]
The /tmp/ovirt-dr-{timestamp}.log would be more reasonable.

**Comment by machacekondra** [comment]
Also to manage the stdout, it's better to use callback plugins:

 https://docs.ansible.com/ansible/2.5/plugins/callback.html#managing-stdout

File: build.sh [hunk 6]
```diff
@@ -44,6 +44,7 @@ install() {
   cp -pR library/ $PKG_DATA_DIR
   cp -pR meta/ $PKG_DATA_DIR
   cp -pR tasks/ $PKG_DATA_DIR
+  cp examples/dr_play.yml $PKG_DATA_DIR
```
Comment (machacekondra) [inline build.sh]: We have this in `doc` directory already, isn't it enought? Or why is it needed also in `usr/share`?

File: examples/dr_play.yml [hunk 7]
```diff
@@ -3,4 +3,4 @@
   hosts: localhost
   connection: local
   roles:
-     - oVirt.disaster-recovery
+      - ovirt-ansible-disaster-recovery
```
Comment (machacekondra) [inline examples/dr_play.yml]: Please use `oVirt.disaster-recovery`

File: files/dr.conf [hunk 8]
```diff
@@ -1,10 +1,14 @@
+[log]
+log_file=tmp2/ovirt-dr-{}.log
+log_level=DEBUG
+
 [generate_vars]
-site=http://engine.example.com/ovirt-engine/api
+site=http://localhost:8080/ovirt-engine/api
```
Comment (machacekondra) [inline files/dr.conf]: please use `engine.example.com`

File: files/fail_over.py [hunk 9]
```diff
@@ -187,4 +224,7 @@ def items(self):
 
 
 if __name__ == "__main__":
-    FailOver().run('dr.conf', '/var/log/ovirt-dr/ovirt-dr.log')
+    level = logging.getLevelName("DEBUG")
+    conf = 'dr.conf'
+    log = '/var/log/ovirt-dr/ovirt-dr.log'
```
Comment (machacekondra) [inline files/fail_over.py]: This directory don't exist so it should be here.

Comment (machacekondra) [reply files/fail_over.py]: It's not, but anyone other testing it, will fail, because he won't have that directory.

File: build.sh (hunk 6 above)
Comment (maorlipchuk) [reply build.sh]: I think that to keep it simple we can also use the dr_play from the examples.
So I will just change the dr.conf to point to the example dr_play which is in the doc directory

**Pull request merged**